### DISCUSSION TITLE
Where does OsmAnd save Navigation routes?

### DISCUSSION THREAD

**Discussion by mr-duke** [osmandapp/OsmAnd#12400] [Q&A]
Dear community,

I would like to know if OsmAnd saves routes that I actively took using the navigation function? If so, where is that data stored?

At first, I was expecting to find it under `/data/media/0/Android/data/net.osmand.plus/files/routing` which would make sense but unfortunately, this folder is empty although I've already taken a few routes.
Just to be precise, I'm not talking about planning a route and saving it a .gpx file (which could be found under `tracks`) but about actual routes using the navigation option.
Also, when starting a new navigation OsmAnd suggests the previous route I took, so I assume that information about navigation has to be saved somewhere.
Your help is highly appreciated!

**Comment by vshcherb** [comment]
I don't think the routes are saved, unless you have trip recording on i.e. they saved in '/tracks/rec'

**Reply by mr-duke** [reply]
@vshcherb You said you think routes are not saved. Can you confirm that or is it just an assumption? Sorry for asking so directly but that piece of information is important for me.
(If routes aren't actually saved I'm wondering where the "Previous route" suggestion when starting a new one comes from.)

**Comment by tommycrock** [comment]
It certainly saves a history of recent destinations (I don't know where this is stored). I think@vscherb means it doesn't save the route itself.

**Comment by mr-duke** [comment]
@tommycrock I see, but what is this directory `/data/media/0/Android/data/net.osmand.plus/files/routing` for ? Judging from the name you would expect it to contain the routes, but no matter what I try, it'll stay empty

Here's my problem: I would like to differentiate between 
a) a user just searches for a destination and 
b ) a user actually went there (by using the Navigation option)
So. that's why I'm interested in reconstructing which routes the user actually tool, i.e which destinations he actually went to.

**Reply by tommycrock** [reply]
No idea. You could try searching the code for routing. I expect what you want is not recorded.

**Reply by Lee-Carre** [reply]
> what is this directory `/data/media/0/Android/data/net.osmand.plus/files/routing` for ?

OsmAnd does offline routing, so maybe (it used to contain) data files for tuning the algorithm for each vehicle / profile.

> I would like to differentiate between …
> …
> I'm interested in reconstructing which routes the user actually tool, i.e which destinations he actually went to.

To what end? Describe your goal, not your current step.

Another approach would be to have the trip-recorder plugin upload to an HTTP server.
